### ISSUE TITLE
Support for Azure Red Hat OpenShift (ARO)

### ISSUE THREAD

**Issue by fedeoliv** [hashicorp/terraform-provider-azurerm#3614]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Provide Azure Red Hat OpenShift (Managed service) ARM template.

### New or Affected Resource(s)

* azurerm_redhatopenshift_cluster

### Potential Terraform Configuration
```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "East US"
}

resource "azurerm_virtual_network" "example" {
  name                = "aro-vnet"
  address_space       = ["10.0.0.0/22"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "master_subnet" {
  name                 = "master-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.0.0/23"]
  service_endpoints    = ["Microsoft.ContainerRegistry"]
  enforce_private_link_endpoint_network_policies = true
}

resource "azurerm_subnet" "worker_subnet" {
  name                 = "worker-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.0.0/23"]
  service_endpoints    = ["Microsoft.ContainerRegistry"]
}

resource "azurerm_redhatopenshift_cluster" "example" {
  name                = "example-redhatopenshift1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  
  # cluster_profile {
  #   pull_secret       = "<my-aroscv.azurecr.io-secret>" 
  #   domain            = "foo.example.com"
  # }

  service_principal {
    client_id     = "00000000-0000-0000-0000-000000000000"
    client_secret = "MyCl1eNtSeCr3t"
  }

  # network_profile {
  #   pod_cidr     = "10.128.0.0/14"
  #   service_cidr = "172.30.0.0/16"
  # }

  master_profile {
    vm_size = "Standard_D8s_v3"
    subnet_id = azurerm_subnet.master_subnet.id
  }
  
  worker_profile {
    # name         = "worker"
    vm_size      = "Standard_D4s_v3"
    subnet_id      = azurerm_subnet.worker_subnet.id
    # disk_size_gb = "128"
    # node_count   = "3"
  }

  # api_server_profile {
  #   visibility = "Public"
  # }

  # ingress_profile {
  #   name       = "default"
  #   visibility = "Public"
  # }

  # tags = {
  #   Environment = "Production"
  # }
}

# output "cluster_version" {
#   value = azurerm_redhatopenshift_cluster.example.version
# }
```

The properties commented are the ones considered as optional.

### References

* https://github.com/Azure/azure-sdk-for-go/blob/main/services/redhatopenshift/mgmt/2020-04-30/redhatopenshift/models.go

**Comment by fernandoBRS** [comment]
There is a WIP here: https://github.com/fernandoBRS/terraform-provider-azurerm/tree/feature/azurerm-managed-openshift

**Comment by r0bnet** [comment]
@fernandoBRS what's the status about the WIP PR? Any updates?

**Comment by SebRosander** [comment]
Any updates on the WIP @fernandoBRS Since the release of 4.3 yesterday this feature is a BIG want :)

**Comment by bentterp** [comment]
I think the APIs are different enought that the new ARO (ocp 4.x) should be another resource

**Comment by lachie-white** [comment]
Man this would be amazing to have in the near future!!!

**Comment by emilianoorengo** [comment]
I need it! The closest approach is using a Deployment template but is not the same. This guy made a lot in this sense: https://github.com/aymenabdelwahed/ARO4-IaC

**Comment by ams0** [comment]
+1, happy to contribute to a working branch if someone has it already

**Comment by erwinsalubre** [comment]
Is there any traction on this?

**Comment by david-yu** [comment]
@mbfrahry Could you close as this now addressed by: https://github.com/hashicorp/terraform-provider-azurerm/pull/24375

**Comment by mbfrahry** [comment]
@david-yu sure can! Fixed with #24375
